File: rocketmq-client/src/main/java/org/apache/rocketmq/client/route/AddressScheme.java
package org.apache.rocketmq.client.route;


import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum AddressScheme {
    DOMAIN_NAME("dns:"),
    IPv4("ipv4:"),
    IPv6("ipv6:");

    public static final String SCHEMA_SEPARATOR = ":";
    private final String prefix;
}

File: csharp/rocketmq-client-csharp/AddressScheme.cs
using rmq = Apache.Rocketmq.V2;

namespace Org.Apache.Rocketmq
{
    public enum AddressScheme
    {
        DomainName,
        Ipv4,
        Ipv6,
    }

    public static class AddressSchemeHelper
    {
        public static rmq.AddressScheme ToProtobuf(AddressScheme scheme)
        {
            switch (scheme)
            {
                case AddressScheme.Ipv4:
                    return rmq.AddressScheme.Ipv4;
                case AddressScheme.Ipv6:
                    return rmq.AddressScheme.Ipv6;
                case AddressScheme.DomainName:
                default:
                    return rmq.AddressScheme.DomainName;
            }
        }
    }
}